library/pk.c

```diff
@@ -238,6 +238,12 @@ int mbedtls_pk_can_do(const mbedtls_pk_context *ctx, mbedtls_pk_type_t type)
 #if defined(MBEDTLS_USE_PSA_CRYPTO)
 /*
  * Tell if a PK can do the operations of the given PSA algorithm
+ *
+ * Note: Key's enrollment is available only when MBEDTLS_PSA_CRYPTO_CLIENT is
+ * defined, i.e. when the Mbed TLS implementation of PSA Crypto is being used.
+ * Even though we don't officially support using other implementations of PSA
+ * Crypto with TLS and X.509 (yet), we're still trying to simplify the life of
+ * people who would like to try it before it's officially supported.
```

Comment:
Minor: I'd prefer this comment to be just before the first `#if defined(MBEDTLS_PSA_CRYPTO_CLIENT)`, in order to be more discoverable and avoiding people getting the wrong impression when reading this #if` without the comment.